internal/pipeline/executor.go

```diff
@@ -283,18 +288,22 @@ func filterMetaData(resources []*yaml.RNode) []*yaml.RNode {
 
 // fnFilters returns chain of functions that are applicable
 // to a given pipeline.
-func fnFilters(_ *Pipeline) []kio.Filter {
-	fn := &annotator{
-		key:   "builtin/setter-1",
-		value: "test",
+func fnFilters(p *Pipeline) ([]kio.Filter, error) {
+	filters, err := p.fnChain()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get function chain: %w", err)
 	}
 
-	// TODO(droot): Implement this with the logic to create
-	// function chain from a pipeline
-	// hardcoding built-in set-annotation function for testing
-	return []kio.Filter{
-		&fnRunner{
-			fn: fn,
-		},
+	var fns []kio.Filter
+	for _, f := range filters {
+		fns = append(fns, f)
 	}
+
+	fns = append(fns, &fnRunner{
+		fn: &annotator{
+			key:   "builtin/setter-1",
+			value: "test",
```

Comment:
are we keeping this function in for testing? is the plan to remove it later?

Reply:
I think we plan to remove it later.

Reply:
I suggest not conflating product code with examples. It makes it hard to follow the purpose of this PR. Instead leave TODO(s) with work that needs to be done in a future PR.